### PR TITLE
[v2.7] Add docker install support for RKE1 custom cluster

### PR DESCRIPTION
**Pull request by rishabhmsra**
With this PR we can pass docker version in `cattle-config` which can be used for installing docker on RKE1 custom cluster nodes.

In order to pass docker version, provide below field in the config:

```
rke1CustomClusterDockerInstall:
  installDockerURL: "https://releases.rancher.com/install-docker/<version-number>.sh"
```

This change is for the AMIs that doesn't have docker pre-installed and are used in docker and OS checks.

### CODE REVIEW

**Review by igomez06 — DISMISSED** [review]
LGTM

File: extensions/provisioninginput/config.go [hunk 1]
```diff
@@ -197,30 +197,35 @@ type NodePools struct {
...
+	Registries                     *Registries                              `json:"registries,omitempty" yaml:"registries,omitempty"`
+	UpgradeStrategy                *rkev1.ClusterUpgradeStrategy            `json:"upgradeStrategy,omitempty" yaml:"upgradeStrategy,omitempty"`
+	Advanced                       *Advanced                                `json:"advanced,omitempty" yaml:"advanced,omitempty"`
+	ClusterSSHTests                []SSHTestCase                            `json:"clusterSSHTests,omitempty" yaml:"clusterSSHTests,omitempty"`
+	CRIDockerd                     bool                                     `json:"criDockerd,omitempty" yaml:"criDockerd,omitempty"`
+	RKE1CustomClusterDockerInstall *RKE1CustomClusterDockerInstall          `json:"rke1CustomClusterDockerInstall,omitempty" yaml:"rke1CustomClusterDockerInstall,omitempty"`
+}
+
+type RKE1CustomClusterDockerInstall struct {
+	InstallDockerURL string `json:"installDockerURL" yaml:"installDockerURL"`
```
Comment (slickwarren) [inline extensions/provisioninginput/config.go:230]: same nit here

**Review by slickwarren — APPROVED** [review]

File: extensions/provisioninginput/config.go (hunk 1 above)
Comment (rishabhmsra) [reply extensions/provisioninginput/config.go]: @slickwarren, I've moved it above config, can i get a quick review on this

**Review by igomez06 — APPROVED** [review]
LGTM

**Review by caliskanugur — APPROVED** [review]
LGTM

**Pull request merged**